### PR TITLE
feat: Add ListChangeStreamPartitions callable

### PR DESCRIPTION
**Pull request by tengzhonger**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigtable/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java [hunk 1]
```diff
@@ -798,6 +805,70 @@ public Map<String, String> extract(ReadModifyWriteRowRequest request) {
...
+                    .build(),
+                settings.listChangeStreamPartitionsSettings().getRetryableCodes());
+
+    ServerStreamingCallable<String, RowRange> userCallable =
+        new ListChangeStreamPartitionsUserCallable(base, requestContext);
+
+    ServerStreamingCallable<String, RowRange> withStatsHeaders =
+        new StatsHeadersServerStreamingCallable<>(userCallable);
+
+    // Copy settings for the middle String -> ByteStringRange callable (as opposed to the inner
```
Comment (mutianf) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java:847]: you mean RowRange?

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java [hunk 2]
```diff
@@ -798,6 +805,70 @@ public Map<String, String> extract(ReadModifyWriteRowRequest request) {
...
+                                "app_profile_id",
+                                listChangeStreamPartitionsRequest.getAppProfileId());
+                          }
+                        })
+                    .build(),
+                settings.listChangeStreamPartitionsSettings().getRetryableCodes());
+
+    ServerStreamingCallable<String, RowRange> userCallable =
+        new ListChangeStreamPartitionsUserCallable(base, requestContext);
+
```
Comment (mutianf) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java:843]: I think you also need `ConvertExceptionCallable` here. Maybe you could refactor ReadRowsConvertExceptionCallable to something like ConvertStreamExceptionCallable

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java (hunk 1 above)
Comment (mutianf) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java:847]: I also don't think this comment is accurate, I think you could just say "Copy idle timeout settings for watchdog"

**Review by mutianf — CHANGES_REQUESTED** [review]

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java (hunk 2 above)
Comment (tengzhonger) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java:842]: Done. Added ConvertStreamExceptionCallable and make ReadRows/ListChangeStreamPartitions callable use it.

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java [hunk 3]
```diff
@@ -1522,6 +1522,7 @@ public UnaryCallable<ReadModifyWriteRow, Row> readModifyWriteRowCallable() {
    *
    * @see ServerStreamingCallable For call styles.
    */
+  @InternalApi
```
Comment (mutianf) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java:1525]: nit: You can also add a comment, something like:

`@InternalApi("Used in dataflow pipeline")`

**Review by mutianf — APPROVED** [review]

**Pull request merged**